### ISSUE TITLE
feat: investigate pre-confirmations on Arbitrum and Optimism for Kakarot ZK-EVM

### ISSUE THREAD

**Issue by Eikix** [kkrt-labs/kakarot-rpc#1173]
# Feature Request

**Describe the Feature Request**

Context: Users want instant feedback. OP blocks are 2 seconds. This is too long for fast UX. Kakarot blocks are longer: ~13s. Way too long! We currently do centralized preconfirmations in the form of "incomplete" transaction receipts that are sent to the users after ~1s or so! They are not supported by many wallets

I've been told that the way Optimism and Arbitrum do pre-confirmations is using a specific RPC endpoint / method for wallets to get fast preconfs on OP or ARB txs.

**Describe Preferred Solution**

- Investigate the way OP and ARB do preconfirmations. Implement a similar logic for Kakarot ZK-EVM at the RPC level.
- Decide if we need to update our current logic for tx receipts.

Note this is a timebox -> result should have:
- description of preconfirmations on ARB/OP
- technical strategy to have feature parity on Kakarot. The endgoal is sub second UX feedback.

**Comment by Eikix** [comment]
Sharing here the latest updates:


### Elias:
Participating in preconfirmation conversations at Zuberlin,
Question to rollups that perform preconfs, afaik Arbitrum and Optimism? 
Is there documentation on those preconfs?

Here in the current L2 meta, preconf is the ability for a centralized sequencer to “promise” a holder that their tx will be included as soon as possible, thus providing subsecond ux feedback instead of ~2 seconds response (== OP block time)


### Ed Felten (Arb tech lead):
In Arbitrum, the sequencer publishes a real-time feed which is a public promise of which transactions will be included and in which order.


The block time in Arbitrum is typically 250 milliseconds.

### Harry Kalodner (CTO Offchain labs):

It doesn’t provide anything faster than the block time. Blocks themselves are issued as pre confirmations when you see them as opposed to waiting for a batch containing those blocks to be posted to Ethereum which takes minutes to accumulate enough transactions to amortize over fixed costs

**Comment by Eikix** [comment]
TL;DR there is no infrastructure to support preconfirmations properly; might close this issue
